File: backend/src/main/java/com/project2/domain/member/dto/MemberDTO.java
package com.project2.domain.member.dto;

import com.project2.domain.member.entity.Member;
import lombok.Getter;
import org.springframework.lang.NonNull;

@Getter
public class MemberDTO {
    @NonNull
    private long id;
    @NonNull
    private String nickname;
    @NonNull
    private String profileImageUrl;

    public MemberDTO(Member member) {
        this.id = member.getId();
        this.nickname = member.getNickname();
        this.profileImageUrl = member.getProfileImageUrlOrDefaultUrl();
    }
}

File: backend/src/main/java/com/project2/domain/member/dto/MemberDTO.kt
package com.project2.domain.member.dto

import com.project2.domain.member.entity.Member

data class MemberDTO(
        val id: Long,
        val nickname: String,
        val profileImageUrl: String
) {
    companion object {
        fun from(member: Member): MemberDTO = MemberDTO(
                id = member.id!!,
                nickname = member.nickname,
                profileImageUrl = member.getProfileImageUrlOrDefaultUrl()
        )
    }
}